### ISSUE TITLE
Begin research on asynchronous training framework

### ISSUE THREAD

**Issue by bgtwosix** [twosixlabs/csl#18]
Begin work on the asynchronous training framework from the proposal. In this framework, two people should be training encoder-decoder models, and occasionally sharing gradients from their own decoder model to the other person's encoder. Start with a trivial example in a notebook just to get the idea started. 

This link may be useful:
https://pytorch.org/tutorials/intermediate/rpc_async_execution.html
